File: pset6/readability/readability.py
from cs50 import get_string
import re

# patterns to detect words, letters and sentences using re module
alphabets = "[a-zA-Z]"
spaces = "[ ]"
end_punctuation = "[!?.]"


def main():
    # propmpt the user for a text
    Text = get_string("Text: ")
    # use grade computer function to calculate the grade level
    grade = grade_computer(Text)
    print(grade)


def grade_computer(Text):

    # counters for leters, words and sentences
    letters_counter, words_counter, sentences_count = 0, 1, 0

    # count number of letters, words, and sentences in the input text
    for i in range(len(Text)):

        if re.search(alphabets, Text[i]):
            letters_counter += 1

        elif re.search(spaces, Text[i]):
            words_counter += 1

        elif re.search(end_punctuation, Text[i]):
            sentences_count += 1

    # calculating the number of letters and sentences per 100 words
    L = letters_counter * 100.0 / words_counter
    S = sentences_count * 100.0 / words_counter
    # computing the grade by Coleman-Liau index formula
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Reading level is under Grade 1
    if index < 1:
        return "Before Grade 1"

    # Reading level is above Grade 16
    if index > 16:
        return "Grade 16+"

    # reading levels between Grade 1 & Grade 16 !
    else:
        return f"Grade {index}"


if __name__ == '__main__':
    main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


//prototyping
int grade_computer(string text_input);

// main function
int main(void)
{
    // prompt the user to get a text!
    string Text;
    do
    {
        Text = get_string("Text: ");
    }

    // No space must be in the beginning!
    while ((Text[0] == '\0') || (Text[0] == ' '));

    // compute the grade!
    int grade = grade_computer(Text);

}




int grade_computer(string text_input)
{

    int letters_counter = 0, words_counter = 1, sentences_count = 0;
    for (int i = 0, n = strlen(text_input); i < n; i++)
    {
        if (isalpha(text_input[i]))
        {
            letters_counter += 1;
        }

        if (isspace(text_input[i]))
        {
            words_counter += 1;
        }

        if ((text_input[i] == '.') || (text_input[i] == '?') || (text_input[i] == '!'))
        {
            sentences_count += 1;
        }
    }

    // calculating the number of letters and sentences per 100 words
    float L = letters_counter * 100.0 / words_counter;
    float S = sentences_count * 100.0 / words_counter;
    // computing the grade by Coleman-Liau index formula
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Reading level is under Grade 1
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;

    }

    // Reading level is above Grade 16
    if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    // reading levels between Grade 1 & Grade 16 !
    else
    {
        printf("Grade %i\n", index);
        return 0;
    }


}


